packages/core/src/web/helpers/fonts/googleFontService.ts

```diff
@@ -0,0 +1,259 @@
+ */
+const loadHistoryGoogleFonts = (): void => {
+  const fontHistory = useStorageStore.getState()['font-history'];
+
+  if (!fontHistory || !Array.isArray(fontHistory) || fontHistory.length === 0) {
+    return;
+  }
+
+  const fontStore = useGoogleFontStore.getState();
+  const networkAvailable = fontStore.isNetworkAvailableForGoogleFonts();
+
+  if (!networkAvailable) {
+    cleanGoogleFontsFromHistory(fontHistory);
+
+    return;
+  }
+
+  const googleFontsFromHistory = fontHistory.filter((family) => !isLocalFont(family));
+
+  if (googleFontsFromHistory.length === 0) {
+    return;
+  }
+
+  googleFontsFromHistory.forEach((family) => {
+    fontStore.loadGoogleFont(family);
+  });
+};
+
+/**
+ * Replace Google Fonts with web-safe fallbacks in SVG text elements
+ */
+const applyGoogleFontFallbacks = (textElements: SVGTextElement[], store: any): void => {
+  let fallbackCount = 0;
+  const replacements: Array<{ from: string; to: string }> = [];
+
+  textElements.forEach((textElem) => {
+    const fontFamily = textElem.getAttribute('font-family');
+
+    if (fontFamily) {
+      const cleanFamily = fontFamily.replace(/^['"]+|['"]+$/g, '').trim();
+
+      if (store.isWebSafeFont(cleanFamily) || isLocalFont(cleanFamily)) {
+        return;
+      }
+
+      const fallbackFont = store.getFallbackFont(cleanFamily);
+      const fallbackPostScriptName = store.getFallbackPostScriptName(fallbackFont);
+
+      textElem.setAttribute('font-family', fallbackFont);
+
+      const postScriptName = textElem.getAttribute('font-postscript-name');
+
+      if (postScriptName) {
+        textElem.setAttribute('font-postscript-name', fallbackPostScriptName);
+      }
+
+      const dataPostScriptName = textElem.getAttribute('data-postscript-name');
+
+      if (dataPostScriptName) {
+        textElem.setAttribute('data-postscript-name', fallbackPostScriptName);
+      }
+
+      const allAttributes = textElem.attributes;
+
+      for (const attr of allAttributes) {
+        if (
+          attr.name.toLowerCase().includes('postscript') &&
+          attr.name !== 'font-postscript-name' &&
+          attr.name !== 'data-postscript-name'
+        ) {
+          textElem.setAttribute(attr.name, fallbackPostScriptName);
+        }
+      }
+
+      const existingReplacement = replacements.find((r) => r.from === cleanFamily);
+
+      if (!existingReplacement) {
+        replacements.push({ from: cleanFamily, to: fallbackFont });
+      }
+
+      fallbackCount++;
+    }
+  });
+
+  if (fallbackCount > 0) {
+    console.log(`Applied ${fallbackCount} Google Font fallbacks during offline import`);
+  }
+};
+
+/**
+ * Load Google Fonts from document context or apply fallbacks when offline
+ */
+export const loadContextGoogleFonts = (): void => {
+  try {
+    const store = useGoogleFontStore.getState();
+    const textElements = [
+      ...document.querySelectorAll('#svgcontent g.layer:not([display="none"]) text'),
```

Comment:
Since this function is only called on loading bvg/beam, file. Maybe `:not([display="none"])` should be removed?
Otherwise changing the layer visibility may broke?